### ISSUE TITLE
MockFileSystem created file don't exist

### ISSUE THREAD

**Issue by milten89** [TestableIO/System.IO.Abstractions#822]
**Describe the bug**
From version 16.0.8 of **System.IO.Abstractions.TestingHelpers** when I create new file in **MockFileSystem** property **Exist** return false.

**To Reproduce**
```cs
IFileSystem fileSystem = new MockFileSystem();
IFileInfo fileInfo = fileSystem.FileInfo.FromFileName("testFile.txt");
using (Stream fs = fileInfo.Create()) 
{
    fs.WriteByte(0);
}
Console.WriteLine($"File exist: {fileInfo.Exists}");
```
Version 16.0.7 return **True**
Version 16.0.8 return **False**

**Expected behavior**
Property Exist should return **True**

**Comment by fgreinacher** [comment]
@milten89 Thanks for taking the time to report this!

The behavior of `MockFileInfo` has changed in #791 to mirror the behavior of the real file system (see #661).

Could you please check how your code behaves with the real file system?

**Comment by milten89** [comment]
@fgreinacher there is difference in output for FileSystem and MockFileSystem.

```cs
public static void Main(string[] args)
{
    IFileSystem realFileSystem = new FileSystem();
    IFileSystem mockFileSystem = new MockFileSystem();
    Console.WriteLine($"Real file exist: {TestFileSystem(realFileSystem)}");
    Console.WriteLine($"Fake file exist: {TestFileSystem(mockFileSystem)}");
    Console.ReadLine();
}

private static bool TestFileSystem(IFileSystem fileSystem)
{
    IFileInfo fileInfo = null; 
    try
    {
        fileInfo = fileSystem.FileInfo.FromFileName("testFile.txt");
        using (Stream fs = fileInfo.Create()) 
            fs.WriteByte(0);

        return fileInfo.Exists;
    }
    finally
    {
        if (fileInfo != null && fileInfo.Exists)
            fileInfo.Delete();
    }
}
```

Starts from version 16.0.8 FileSystem return **True** and MockFileSystem return **False**.
In current latest version 16.1.23 result is same.

Additional to that, there is difference in paths for FileSystem and MockFileSystem.
- FileSystem: 
```
C:\Users\Andrzej Ochodek\source\repos\FileSystemAbstractionTest\FileSystemAbstractionTest\bin\Debug\testFile.txt
```
- MockFileSystem: 
```
C:\testFile.txt
```

**Comment by milten89** [comment]
Are there any updates about this bug?

**Comment by milten89** [comment]
> Do I understand correctly that everything is working as expected in 16.1.23?

Unfortunately no. Real file system returns _true_ and mock returns _false_.

> That's expected/by-design, because the MockFileSystem is initialized with the root directory (`/` on Unix, `C:\` on Windows).

Ok.

**Comment by siprbaum** [comment]
I created a pull request to fix it, see #828
